VEDA_OS01/management/commands/tests/test_create_oauth_app_client.py

```diff
@@ -0,0 +1,185 @@
+        (dict(zip(('username', 'redirect_uris', 'client_name', 'client_secret', 'skip_authorization'), p)) for p in product(
+            (USERNAME, None),
+            (REDIRECT_URI, None),
+            ('client_name', None),
+            ('client_secret', None),
+            (True, False)
+        ))
+    ))
+    @ddt.unpack
+    def test_client_creation(self, args, options):
+        """
+        Verify that the command creates a Client when given valid arguments and options.
+        """
+        self.assert_client_created(args , options)
+
+    @ddt.data(
+        ((CLIENT_ID,), 'too few arguments'),
+        ((CLIENT_ID, REDIRECT_URI, CLIENT_TYPES[0], CLIENT_TYPES[1]), 'unrecognized arguments'),
+    )
+    @ddt.unpack
+    def test_argument_cardinality(self, args, err_msg):
+        """
+        Verify that the command fails when given an incorrect number of arguments.
+        """
+        with self.assertRaises(CommandError) as exc:
+            self._call_command(args, {})
+
+        self.assertIn(err_msg, exc.exception.message)
+
+    def test_client_type_validation(self):
+        """
+        Verify that the command fails when the provided client type is invalid.
+        """
+        with self.assertRaises(CommandError) as exc:
+            self._call_command((CLIENT_ID, 'invalid_client_type', GRANT_TYPES[0]))
+
+        self.assertEqual(
+            "Client type provided is invalid. Please use one of {}.".format(CLIENT_TYPES),
+            exc.exception.message
+        )
+
+    def test_grant_type_validation(self):
+        """
+        Verify that the command fails when the provided grant type is invalid.
+        """
+        with self.assertRaises(CommandError) as exc:
+            self._call_command((CLIENT_ID, CLIENT_TYPES[0], 'invalid_grant_type'))
+
+        self.assertEqual(
+            "Grant type provided is invalid. Please use one of {}.".format(GRANT_TYPES),
+            exc.exception.message
+        )
+
+    def test_username_validation(self):
+        """
+        Verify that the command fails when the provided username is invalid.
+        """
+        with self.assertRaises(CommandError) as exc:
+            self._call_command(
+                (CLIENT_ID, CLIENT_TYPES[0], GRANT_TYPES[0]),
+                {'username': 'invalid'}
+            )
+
+        self.assertEqual(
+            'User matching the provided username does not exist.',
+            exc.exception.message
+        )
+
+    def test_url_validation(self):
+        """
+        Verify that the command fails when the provided URLs are invalid.
+        """
+        args = CLIENT_ID, CLIENT_TYPES[0], GRANT_TYPES[0]
+        with self.assertRaises(CommandError) as exc:
+            self._call_command(args, {'redirect_uris': 'invalide uri'})
+
+        self.assertEqual(
+            'URIs provided are invalid. Please provide valid redirect URIs.',
+            exc.exception.message
+        )
+
+    def test_idempotency(self):
```

Comment:
Can we check the idempotency via `ddt`?

Reply:
I don't think so. I will prefer to leave this as it is.